src/domain/login/PasswordLoginViewModel.js

```diff
@@ -39,10 +41,35 @@ export class PasswordLoginViewModel extends ViewModel {
         this._errorMessage = message;
         this.emitChange("errorMessage");
     }
+    
+    setLoginOptions(loginOptions) {
+		this._loginOptions = loginOptions;
+	}
+	
+	setEnabled(enabled) {
+		this._enabled = enabled;
+        this.emitChange("isEnabled");
+	}
 
+    async parseUsernameLogin (login) {
+        if ((login.match(/@/g) || []).length == 1 && (login.match(/:/g) || []).length == 1 && (login.match(/\./g) || []).length) {
```

Comment:
Consider using just one regex to parse the mxid and extract the server name : `@.*:(.+)`

Reply:
So I replace the 55th line's condition by `login.match(/@.*:(.+)/)`?